python/ray/ml/examples/tensorflow/tensorflow_linear_dataset_example.py

```diff
@@ -0,0 +1,149 @@
+import argparse
+import numpy as np
+
+
+import tensorflow as tf
+from tensorflow.keras.callbacks import Callback
+
+import ray
+import ray.train as train
+from ray.data import Dataset
+from ray.data.dataset_pipeline import DatasetPipeline
+from ray.train.tensorflow import prepare_dataset_shard
+from ray.ml.train.integrations.tensorflow import TensorflowTrainer
+from ray.ml.predictors.integrations.tensorflow import TensorflowPredictor
+from ray.ml.constants import MODEL_KEY, TRAIN_DATASET_KEY
+from ray.ml.result import Result
+
+
+class TrainCheckpointReportCallback(Callback):
+    def on_epoch_end(self, epoch, logs=None):
+        train.save_checkpoint(**{MODEL_KEY: self.model.get_weights()})
+        train.report(**logs)
+
+
+def get_dataset_pipeline(a=5, b=10, size=1000) -> DatasetPipeline:
+    def get_dataset(a, b, size) -> Dataset:
+        items = [i / size for i in range(size)]
+        dataset = ray.data.from_items([{"x": x, "y": a * x + b} for x in items])
+        return dataset
+
+    dataset = get_dataset(a, b, size)
+    train_dataset_pipeline = dataset.repeat().random_shuffle_each_window()
+
+    return train_dataset_pipeline
+
+
+def build_model() -> tf.keras.Model:
+    model = tf.keras.Sequential(
+        [
+            tf.keras.layers.InputLayer(input_shape=(1,)),
+            tf.keras.layers.Dense(10),
+            tf.keras.layers.Dense(1),
+        ]
+    )
+    return model
+
+
+def train_func(config: dict):
+    batch_size = config.get("batch_size", 64)
+    epochs = config.get("epochs", 3)
+
+    strategy = tf.distribute.MultiWorkerMirroredStrategy()
+    with strategy.scope():
+        # Model building/compiling need to be within `strategy.scope()`.
+        multi_worker_model = build_model()
+        multi_worker_model.compile(
+            optimizer=tf.keras.optimizers.SGD(learning_rate=config.get("lr", 1e-3)),
+            loss=tf.keras.losses.mean_squared_error,
+            metrics=[tf.keras.metrics.mean_squared_error],
+        )
+
+    dataset_pipeline = train.get_dataset_shard(TRAIN_DATASET_KEY)
```

Comment:
Same thing here- can we just change this to "train" instead of TRAIN_DATASET_KEY?